### PR TITLE
[MM-58509] Migrate tooltips of "components/admin_console/user_grid/user_grid_role_dropdown.tsx" to WithTooltip

### PR DESCRIPTION
**Pull request by Aryakoste**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/27237
Jira https://mattermost.atlassian.net/browse/MM-58509

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/admin_console/user_grid/user_grid_role_dropdown.tsx [hunk 1]
```diff
@@ -133,27 +131,23 @@ export default class UserGridRoleDropdown extends React.PureComponent<Props> {
...
-                    />
-                </Tooltip>
-            );
-
             return (
                 <div className='more-modal__shared-actions'>
-                    <OverlayTrigger
-                        delayShow={Constants.OVERLAY_TIME_DELAY}
+                    <WithTooltip
+                        id='shared_user_indicator.tooltip'
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/user_grid/user_grid_role_dropdown.tsx:137]:
```suggestion
                        id='sharedUserIndicator.Tooltip'
```

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/admin_console/user_grid/user_grid_role_dropdown.tsx [hunk 2]
```diff
@@ -133,27 +131,23 @@ export default class UserGridRoleDropdown extends React.PureComponent<Props> {
...
                         placement='bottom'
-                        overlay={sharedTooltip}
+                        title={
+                            <FormattedMessage
+                                id='shared_user_indicator.tooltip.text'
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/user_grid/user_grid_role_dropdown.tsx:141]: lets not change the translation

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/admin_console/user_grid/user_grid_role_dropdown.tsx (hunk 1 above)
Comment (M-ZubairAhmed) [reply webapp/channels/src/components/admin_console/user_grid/user_grid_role_dropdown.tsx]: @Aryakoste we missed this one

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Review by yasserfaraazkhan — CHANGES_REQUESTED** [review]
@M-ZubairAhmed @Aryakoste 

the screen crashes when we click on users in the System Console > Users page

Screenshot 2024-06-05 130324

**Comment by M-ZubairAhmed** [comment]
> @M-ZubairAhmed @Aryakoste
> 
> the screen crashes when we click on users in the System Console > Users page
> 
> Screenshot 2024-06-05 130324

This should be fixed by https://github.com/mattermost/mattermost/pull/27419

**Review by yasserfaraazkhan — APPROVED** [review]

**Pull request merged**